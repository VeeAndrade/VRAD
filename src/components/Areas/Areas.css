.main-areas {
  height: 100%;
}

.heading-container-wrapper {
  display: flex;
  height: 7%;
  justify-content: center;
  align-content: center;
}

.card-container {
  display: grid;
  height: 60%;
  grid-template-columns: 1fr 1fr;
  /* grid-gap: 10px; */
  justify-items: center;
  text-align: left;
  margin-top: 2%;
}

.heading {
  width: 100%;
  background-color: #fff;
  color: #007EA7;
}