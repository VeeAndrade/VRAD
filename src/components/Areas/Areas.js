import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import AreaCard from '../AreaCard/AreaCard';
import PropTypes from 'prop-types'
import './Areas.css'

class Areas extends Component {
  constructor() {
    super()
    this.state = {
      areas: ''}
  }

  componentDidMount() {
    this.setState({areas: this.props.areasData})
  }

  displayAreas = () => {
    return this.state.areas.map(area => {
      return <AreaCard
      shortName={area.shortName}
      name={area.name}
      description={area.about}
      areaId={area.id}
      key={area.id}
      />
    })
  }

  render() {
    if(!this.state.areas) {
      return <h1>Loading...</h1>
    } else {
      return(
        <main className='main-areas'>
          <section className='heading-container-wrapper'>
            <h2 className='heading'>Denver Areas</h2>
            <NavLink className='signout-btn' to='/'>Signout</NavLink>
          </section>
          <section className='card-container'>
            {this.displayAreas()}
          </section>
        </main>
      )
    }
  }
}

Component.propTypes = {
  areas: PropTypes.array
}

export default Areas;
