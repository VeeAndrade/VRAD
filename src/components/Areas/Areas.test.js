import React from 'react';
import { shallow } from 'enzyme'
import Areas from './Areas';

describe('Areas', () => {
  let wrapper;
  const mockAreasData = [{
      area: "RiNo",
      details: '/api/v1/areas/590'
    },
    {
      area: "Park Hill",
      details: '/api/v1/areas/751'
    },
    {
      area: "LoHi",
      details: '/api/v1/areas/408'
    },
    {
      area: "Cap Hill",
      details: '/api/v1/areas/240'
    }
  ] 
  
  beforeEach(() => {
    wrapper = shallow(<Areas areasData={mockAreasData} />)
  });

  it('should match the areas snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should have a default state', () => {
    expect(wrapper.state('areas')).toEqual(mockAreasData)
    expect(wrapper.state('listings')).toEqual('')
    expect(wrapper.state('userFavorites')).toEqual('')
  });
})