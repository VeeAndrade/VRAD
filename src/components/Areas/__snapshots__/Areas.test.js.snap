// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Areas should match the areas snapshot 1`] = `
<main
  className="main-areas"
>
  <section
    className="heading-container-wrapper"
  >
    <h2
      className="heading"
    >
      Denver Areas
    </h2>
    <NavLink
      className="signout-btn"
      to="/"
    >
      Signout
    </NavLink>
  </section>
  <section
    className="card-container"
  >
    <AreaCard />
    <AreaCard />
    <AreaCard />
    <AreaCard />
  </section>
</main>
`;
